src/use-dropdown-menu.ts

```diff
@@ -1,6 +1,15 @@
 // Imports
 import React, { useState, useRef, createRef, useEffect } from 'react';
 
+// Create interface for button properties
+interface ButtonProps
+	extends Pick<
+		React.DetailedHTMLProps<React.ButtonHTMLAttributes<HTMLButtonElement>, HTMLButtonElement>,
+		'onKeyDown' | 'onClick' | 'tabIndex' | 'role' | 'aria-haspopup' | 'aria-expanded'
+	> {
+	ref: ((instance: HTMLButtonElement | null) => void) | React.RefObject<HTMLButtonElement> | null | undefined;
```

Comment:
Is it not possible to use `Pick<>` to pick out the `ref` prop too?

Reply:
You can, but, for some reason the definition of ref is different from the definition that this issue arose from. I meant to start a discussion about that on this PR. 

I tracked down the type in JSX.IntrisicValues for button, it is identical to the value we are using to define our props, but for some reason ref seems to be defined without `string` as a possible type for ref, but in our hook string is a possible type if we define it using `React.DetailedHTMLProps<React.ButtonHTMLAttributes<HTMLButtonElement>, HTMLButtonElement>`. Since in our implementation the ref is never defined with a type of string, this was one possible solution.